packages/hub/src/lib/LazyBlob.ts

```diff
@@ -0,0 +1,84 @@
+import { createReadStream } from "fs";
+import { open } from "fs/promises";
+import type { FileHandle } from "fs/promises";
+
+export class LazyBlob {
+	private path: string;
+	private file: FileHandle | null;
+	private totalSize: number;
+
+	constructor(path: string) {
+		this.path = path;
+
+		this.file = null;
+		this.totalSize = 0;
+	}
+
+	get size(): number {
+		return this.totalSize;
+	}
+
+	get lentgh(): number {
+		return this.size;
+	}
+
+	get type(): string {
+		return "";
+	}
+
+	async init(): Promise<void> {
+		this.file = await open(this.path, "r");
+
+		const { size } = await this.file.stat();
+		this.totalSize = size;
+	}
+
+	async dispose(): Promise<void> {
+		if (this.file === null) {
+			return;
+		}
+
+		await this.file.close();
+	}
+
+	async slice(start: number, end: number): Promise<Blob> {
+		if (this.file === null) {
+			throw new Error("LazyBlob has not been initialized");
+		}
+
+		const size = end - start;
```

Comment:
About negative values for `end` and `start`, althought it's possible to do this, it's not trivial because it would require in some case to read the file 2 times. 

For example on a file of size 10:
 - `slice(5, -2)`: read from 5 to 10, then from 0 to 1
 
To be honest, I'm not sure it's necessary to mimic the whole `Blob` class since we could just stick to the usecase we have. If the feature is needed later then we can add it at the cost of more complexity .